gobblin-example/src/main/java/gobblin/example/wikipedia/WikipediaExtractor.java

```diff
@@ -162,15 +250,21 @@ private boolean isPropPresent(String key, WorkUnitState workUnitState) {
         || workUnitState.getJobState().contains(key);
   }
 
-  private Queue<JsonElement> retrievePageRevisions(String pageTitle) throws IOException {
-    Queue<JsonElement> retrievedRevisions = new LinkedList<>();
+  private JsonElement performHttpQuery(String rootUrl, Map<String, String> query) throws URISyntaxException, IOException {
+
+    List<String> queryTokens = Lists.newArrayList();
+    for (Map.Entry<String, String> entry : query.entrySet()) {
+      queryTokens.add(entry.getKey() + "=" + entry.getValue());
+    }
+    String actualQuery = Joiner.on("&").skipNulls().join(queryTokens);
+
+    URL actualURL = new URL(rootUrl + (rootUrl.contains("?") ? "&" : "?") + actualQuery);
```

Comment:
This looks dangerous. From the javadocs of the URL class:

> The URL class does not itself encode or decode any URL components according to the escaping mechanism defined in RFC2396

I suggest using org.apache.http.client.utils.URIBuilder or some such for proper escaping.